"AttributeError: object has no attribute 'add_mark'" when marks clash with existing attributes

When running the following code with `python -m pytest foo.py`:

```python
import pytest


def assign_foo(f):
    f.foo = "Hi"
    return f


@pytest.mark.foo
@assign_foo
def test():
    pass
```

I get the following error:

```
foo.py:10: in <module>
    @assign_foo
v/lib/python3.6/site-packages/_pytest/mark/structures.py:213: in __call__
    store_legacy_markinfo(func, self.mark)
v/lib/python3.6/site-packages/_pytest/mark/structures.py:250: in store_legacy_markinfo
    holder.add_mark(mark)
E   AttributeError: 'str' object has no attribute 'add_mark'
```

This is on 3.6.0 with the following packages in the virtualenv (i.e. a clean install of pytest 3.6.2):

```
Package        Version
-------------- -------
atomicwrites   1.1.5
attrs          18.1.0
more-itertools 4.2.0
pip            10.0.1
pluggy         0.6.0
py             1.5.4
pytest         3.6.2
setuptools     39.2.0
six            1.11.0
wheel          0.31.1
```

We encountered this because someone was applying `@pytest.mark.hypothesis` to a `@given` decorated test (see https://github.com/HypothesisWorks/hypothesis/issues/1362), which broke when we added a `hypothesis` attribute on the decorated function. 

I believe this is only a problem with the legacy support and is not required for marks to work: we already apply a hypothesis mark to all `@given` decorated tests, and it is still working fine, so it seems likely that a fix that checked if the attribute was already present and did nothing if it was not a `MarkInfo` would be backwards compatible and fix this problem.

--- a/changelog/3631.bugfix.rst
+++ b/changelog/3631.bugfix.rst
@@ -1,0 +1,1 @@
+No longer raise AttributeError when legacy marks can't be stored.

--- a/src/_pytest/mark/structures.py
+++ b/src/_pytest/mark/structures.py
@@ -259,7 +259,7 @@ def store_legacy_markinfo(func, mark):
     if holder is None:
         holder = MarkInfo.for_mark(mark)
         setattr(func, mark.name, holder)
-    else:
+    elif isinstance(holder, MarkInfo):
         holder.add_mark(mark)
 
 

--- a/testing/test_mark.py
+++ b/testing/test_mark.py
@@ -63,6 +63,19 @@ class TestMark(object):
         mark.hello(f)
         assert f.hello
 
+    def test_mark_legacy_ignore_fail(self):
+        def add_attribute(func):
+            func.foo = 1
+            return func
+
+        @pytest.mark.foo
+        @add_attribute
+        def test_fun():
+            pass
+
+        assert test_fun.foo == 1
+        assert test_fun.pytestmark
+
     @ignore_markinfo
     def test_pytest_mark_keywords(self):
         mark = Mark()

